contrib/node/tests/python/pants_test/contrib/node/tasks/test_apache_thrift_node_gen.py

```diff
@@ -0,0 +1,87 @@
+# import os
+from textwrap import dedent
+
+# from pants.backend.codegen.thrift.lib.thrift import Thrift
+from pants.contrib.node.tasks.apache_thrift_node_gen import ApacheThriftNodeGen
+from pants.contrib.node.targets.node_thrift_library import NodeThriftLibrary
+from pants.contrib.node.targets.node_module import NodeModule
+from pants_test.task_test_base import TaskTestBase
+
+
+class ApacheThriftNodeGenTest(TaskTestBase):
+
+  @classmethod
+  def task_type(cls):
+    return ApacheThriftNodeGen
+
+  def generate_single_thrift_target(self, node_thrift_library):
+    context = self.context(target_roots=[node_thrift_library])
+    apache_thrift_gen = self.create_task(context)
+    apache_thrift_gen.execute()
+    # import pdb; pdb.set_trace()
+
+    def is_synthetic_node_library(target):
+      return isinstance(target, NodeModule) and target.is_synthetic
+
+    synthetic_targets = context.targets(predicate=is_synthetic_node_library)
+
+    self.assertEqual(1, len(synthetic_targets))
+    return synthetic_targets[0]
+
+  def test_single_namespace(self):
+    self.create_file('src/thrift/com/foo/test.thrift', contents=dedent("""
+    namespace js foo.bar
+
+    struct Test {}
+    """))
+    test = self.make_target(spec='src/thrift/com/foo:one',
+                            target_type=NodeThriftLibrary,
+                            sources=['test.thrift'])
+    synthetic_target = self.generate_single_thrift_target(test)
+    self.assertEqual({'package.json',
+                      'yarn.lock',
+                      'test_types.js'},
+                     set(synthetic_target.sources_relative_to_source_root()))
```

Comment:
package.json and yarn.lock should be in the workdir rather than relative from source root

Reply:
so just check in `synthetic_target.sources_relative_to_target_base()`